pallets/aleph/src/lib.rs

```diff
@@ -165,11 +170,33 @@ pub mod pallet {
             }
         }
 
-        pub(crate) fn update_authorities(
-            authorities: &[T::AuthorityId],
-            next_authorities: &[T::AuthorityId],
-        ) {
-            <Authorities<T>>::put(authorities);
+        fn get_authorities_for_next_session(
+            next_authorities: Vec<(&T::AccountId, T::AuthorityId)>,
+        ) -> Vec<T::AuthorityId> {
+            let next_committee_ids = NextFinalityCommittee::<T>::take();
+
+            let get_authority = |cid| {
+                next_authorities
+                    .iter()
+                    .find(|(id, _)| *id == cid)
+                    .map(|(_, auth)| auth.clone())
+            };
+            let next_committee_authorities: Vec<_> = next_committee_ids
+                .iter()
+                .filter_map(get_authority)
+                .collect();
+
+            if next_committee_authorities.len() != next_committee_ids.len() {
+                warn!(target: LOG_TARGET, "Not all committee members were converted to keys. Falling back to the default committee");
+            }
+
+            next_committee_authorities
```

Comment:
I think there is a way to 1) make the time complexity better than O(|next_authorities|*|next_finality_authorities|), 2) avoid cloning anything.

For that I think you can do the following:
1) Turn `next_committee_ids` into a HashSet `set` (or, if not possible in the runtime, use BTreeSet). 
2) Iterate over next_authorities, if `set` contains `id`, then take the authority to `next_committee_authorities`. Otherwise ignore. You can use `set.remove(id)` to avoid cloning.

Reply:
done